### ISSUE TITLE
The whole page "Orders" is redraw when you sort the orders list

### ISSUE THREAD

**Issue by fabienvallon** [PrestaShop/PrestaShop#27018]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

The whole page is redraw when you sort orders list

### Expected behavior

Only the list should be redraw

### Steps to reproduce

1. Go to Orders 
2. Sort the list ( for example customers ) 
3. The whole page is redraw

### PrestaShop version(s) where the bug happened

1.7.8.2

**Comment by matks** [comment]
Hello @fabienvallon the full redraw is not a bug it's a feature.

You're thinking that it's useless to redraw the whole page, and you are right. You're thinking we could send an ajax query; get the data (here, the list content with the sort applied) and redraw only the list.

But if we do this, then the list becomes drawable in 2 usecases at least:
- when the page loads for the 1st time, on synchronous HTTP request
- when it is refreshed, on ajax query

As maintainers we can do it, no issue about that. But this means that any module that is hooked on this page and extend/decorate this listing must also implement the 2 usecases.

Today, as the page is loaded in a simple and unefficient manner, it's very simple for modules to decorate the list. Add a column, modify a row, customize how it's rendered... the module can modify either the data, or the rendering performed by twig, or both.

If we introduce an ajax refresh, it means more work for module developers who must think their module and how they extend PrestaShop in 2 contexts: with and without ajax.

So my opinion is that this basic "redraw the whole page" behavior, which can look like a bug from the outside, is actually a clear decision made to make extending the software easier.

You will find other behaviors in the software where the system has been kept simple and limited in order to make it easy to customize, as PrestaShop is a CMS.

**Comment by fabienvallon** [comment]
OK.
Thx
